tensorly/tenalg/svd.py

```diff
@@ -39,7 +39,7 @@ def svd_flip(U, V, u_based_decision=True):
         )
         U = U * signs
         if tl.shape(V)[0] > tl.shape(U)[1]:
-            signs = tl.concatenate((signs, tl.ones(tl.shape(V)[0] - tl.shape(U)[1])))
+            signs = tl.concatenate((signs, tl.ones(tl.shape(V)[0] - tl.shape(U)[1], device=signs.device)))
```

Comment:
We can directly pass `**tl.context(signs)` instead of just the device so the type is also preserved.